src/lsptoolshost/languageStatusBar.ts

```diff
@@ -40,3 +41,43 @@ class WorkspaceStatus {
         });
     }
 }
+
+class ProjectContextStatus {
+    static createStatusItem(
+        context: vscode.ExtensionContext,
+        languageServer: RoslynLanguageServer,
+        languageServerEvents: RoslynLanguageServerEvents
+    ) {
+        const projectContextService = languageServer._projectContextService;
+
+        const item = vscode.languages.createLanguageStatusItem(
+            'csharp.projectContextStatus',
+            languageServerOptions.documentSelector
+        );
+        item.name = vscode.l10n.t('C# Project Context Status');
+        item.detail = vscode.l10n.t('Project Context');
+        context.subscriptions.push(item);
+
+        updateItem(vscode.window.activeTextEditor);
+        context.subscriptions.push(vscode.window.onDidChangeActiveTextEditor(updateItem));
```

Comment:
I presume we can only have one active text editor right?  If so we should like have a cancellation source here where the prior request gets cancelled every time this changes.

e.g. similar to https://github.com/dotnet/vscode-csharp/blob/main/src/lsptoolshost/onAutoInsert.ts#L66

Reply:
Updated.